enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/outcome/Outcome.java

```diff
@@ -41,27 +41,27 @@
 public class Outcome<MEMBER> implements Serializable
 {
     /* Common */
-    public Role newRole;
+    private Role newRole;
 
-    public long term;
-    public MEMBER leader;
+    private long term;
+    private MEMBER leader;
 
-    public long leaderCommit;
+    private long leaderCommit;
 
-    public ArrayList<LogCommand> logCommands = new ArrayList<>();
-    public ArrayList<RaftMessages.Directed<MEMBER>> outgoingMessages = new ArrayList<>();
+    private ArrayList<LogCommand> logCommands = new ArrayList<>();
```

Comment:
Can we please use interfaces in the declaration side like `List<LogCommand>` and `Set<MEMBER>`?
It is also better to expose interfaces from getters and setters.